Resin Basin updated. Thanks to Spearkiller and Ayutac.

Ayutac opened the pull request:
Hitbox has been updated to half a block, better matching the model.
Item model has been updated by Spearkiller to look nice.
There was a weird inconsistency about the amount of sap given to player depending on either putting it into an inventory or breaking the block.

Right click to harvest basin didn't work as intended, so removed for now.

Also it seems the block states (full/pouring/pouringTime) are not saved via nbt?

Also also wow the warnings.

Commit:
Resin Basin updated. Thanks to Spearkiller and Ayutac.

Hitbox has been updated to half a block, better matching the model.
Item model has been updated by Spearkiller to look nice.
There was a weird inconsistency about the amount of sap given to player depending on either putting it into an inventory or breaking the block.

Right click to harvest basin didn't work as intended, so removed for now.

Also it seems the block states (full/pouring/pouringTime) are not saved via nbt?

Also also wow the warnings.

modmuss50 commented on src/main/java/techreborn/blocks/machine/tier1/ResinBasinBlock.java:
```diff
@@ -82,6 +88,19 @@ public Direction getFacing(BlockState state) {
 
+	/*@SuppressWarnings("deprecation")
+	@Override
+	public ActionResult onUse(BlockState state, World world, BlockPos pos, PlayerEntity player, Hand hand, BlockHitResult hit) {
```
Is this needed? Generally dont want to commit commented out code?

Ayutac replied:
The idea is that it *should* work, but it doesn't. I don't want to commit a bug, but I want people to be able to solve the bug, because I am unlikely to follow it further.

Commit:
Added removal of sap from basin when right -clicking.

The error was me not understanding that "setFullState" had nothing to do with the variable "isFull". My bad, my bad.

For some reason one right click triggers the onUse twice, and previously I didn't correctly update the sap after the first one, but now it should work. Testing went off without a hitch.

Ayutac commented:
I fixed the out-commented bug by myself now :)
